text/0000-embroider-v2-package-format.md

```diff
@@ -0,0 +1,772 @@
+- Start Date: 2019-06-21
+- Relevant Team(s): Ember CLI, Ember.js, Learning
+- RFC PR:
+- Tracking:
+
+# v2 Addon Format (Embroider Compatibility)
+
+## Summary
+
+This RFC defines a new package format that is designed to make Ember packages (meaning both apps and addons) statically analyzable and more compatible with the rest of the NPM & Javascript ecosystem. This RFC is the first step in stabilizing [Embroider](https://github.com/embroider-build/embroider) as our next-generation build system.
+
+## Motivation
+
+One of the good things about Ember is that apps and addons have a powerful set of build-time capabilities that allow lots of shared code with zero-to-no manual integration steps for the typical user. We have been doing “zero config” since before it was a cool buzzword (it was just called “convention over configuration”). And we’ve been broadly successful at maintaining very wide backward- and forward-compatibility for a large body of highly-rated community-maintained addons.
+
+But one of the challenging things about Ember is that our ecosystem’s build-time capabilities are more implementation-defined than spec-defined, and the implementation has accumulated capabilities organically while only rarely phasing out older patterns. I believe the lack of a clear, foundational, build-time public API specification is the fundamental underlying issue that efforts like the various packaging / packager RFCs have tried to work around.
+
+The benefits to users for this RFC (and Embroider in general) are:
+
+- faster builds and faster NPM installs
+- “zero-config import from NPM — both static and dynamic” as a first-class feature that works for both third-party libraries and Ember addons
+- support for arbitrary code splitting
+- tree-shaking of unused modules, components, helpers, etc from the app and all addons
+- a layered build system with clearly documented APIs between the layers, so it's easier to experiment and contribute
+- a build system that can take advantage of current and future investments by the wider Javascript ecosystem into code bundling & optimization.
+
+## Key Ideas
```

Comment:
This section would benefit from a link to an existing, Embroidered Ember app or two (Octane + Super Rentals) and an addon (???). The necessary detail in this RFC makes it sound very complicated to use Embroider, while the median experience is pretty straightforward.